### ISSUE TITLE
include price of instance by region

### ISSUE THREAD

**Issue by kapilt** [garnaat/missingcloud#6]
another 'missing' bit of useful of info.

**Comment by ehammond** [comment]
+1

Might as well toss in the costs of the various Reserved Instances (light, medium, heavy) and their respective hourly charges.

**Comment by garnaat** [comment]
I agree that would be useful.  It's on my list, however I should warn you it's a very long list 8^)

**Comment by jtriley** [comment]
For those interested there is an (unofficial?) set of json files that Amazon uses to maintain it's instance type pricing page. You'll have to look at the source to find them but for example the on-demand linux pricing by region is available here:

http://aws.amazon.com/ec2/pricing/json/linux-od.json

There are similar files for spot and reserved instances. Funny enough this file is also the best source of info for which instance types are available in a given region. It would be nice if Amazon maintained this repo or something similar and actually updated it when new service details come out given that they're clearly doing this for their website/aws console anyway.

**Comment by sopel** [comment]
@jtriley - Thanks for the details regarding these JSON sources, which are also used by @srossross's pull request #27 referenced above -  and I agree, it's a bit odd this available information isn't, well, made officially available (via an API).

In general I think this project would need a tiny build script to allow updating retrievable information like this once in a while; however, before embarking into that direction, we should clarify whether this project might be obsolete anytime soon eventually due to all the work @garnaat has put into the [AWS Command Line Interface](http://aws.amazon.com/cli/) resp. the underlying library [botocore](https://github.com/boto/botocore), where many service details are provided via JSON files already (supposedly generated from upstream API documentation/schemata?!), including an extension mechanism for local data (where e.g. pricing might fit in as well).

**Comment by garnaat** [comment]
I am going to try to get an official feed for this data setup from AWS.  I can't guarantee that I will be successful and I'm not sure how long it might take but that is my goal.  This seems like valuable information and having a dependable, authoritative source is the best way to help customers take advantage of it.

I'll keep you posted.
